### ISSUE TITLE
sentry: processors.go:851: %s already closed | string

### ISSUE THREAD

**Issue by tbg** [cockroachdb/cockroach#32116]
https://sentry.io/cockroach-labs/cockroachdb/issues/749898688/

processors.go:851: %s already closed | string

github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*RowChannel).ConsumerClosed

> stacktrace: {u'frames': [{u'function': u'Run', u'abs_path': u'/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go', u'module': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*sortTopKProcessor)', u'filename': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go', u'lineno': 489, u'in_app': False}, {u'function': u'Run', u'abs_path': u'/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go', u'module': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun', u'filename': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go', u'lineno': 170, u'in_app': False}, {u'function': u'ConsumerClosed', u'abs_path': u'/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go', u'module': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*sortTopKProcessor)', u'filename': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go', u'lineno': 511, u'in_app': False}, {u'function': u'close', u'abs_path': u'/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go', u'module': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*sortTopKProcessor)', u'filename': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go', u'lineno': 499, u'in_app': False}, {u'function': u'ConsumerClosed', u'abs_path': u'/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go', u'module': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*RowChannel)', u'filename': u'github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go', u'lineno': 446, u'in_app': False}]}
type: *log.safeError
value: processors.go:851: %s already closed | string

**Comment by jordanlewis** [comment]
2.1 alpha.

```
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go in ConsumerClosed at line 446
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go in close at line 499
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go in ConsumerClosed at line 511
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go in Run at line 170
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go in Run at line 489
```

**Comment by jordanlewis** [comment]
We shouldn't panic on double close, even though it's wrong to do it.

**Comment by jordanlewis** [comment]
Was closed by #35125. @jordanlewis to backport that PR.
